### PR TITLE
Qualify error throwing functions as noreturn

### PR DESCRIPTION
**Pull request by nordlow**

### CODE REVIEW

File: druntime/src/core/exception.d [hunk 1]
```diff
@@ -608,7 +608,7 @@ extern (C) void onUnittestErrorMsg( string file, size_t line, string msg ) nothr
  * Throws:
  *  $(LREF RangeError).
  */
-extern (C) void onRangeError( string file = __FILE__, size_t line = __LINE__ ) @trusted pure nothrow @nogc
+extern (C) void onRangeError( string file = __FILE__, size_t line = __LINE__ ) @trusted pure nothrow @nogc noreturn
```
Comment (dkorpel) [inline druntime/src/core/exception.d:611]: `noreturn` is a type, not a function attribute

Comment (nordlow) [reply druntime/src/core/exception.d:611]: Ahh, will adjust.

Comment (nordlow) [reply druntime/src/core/exception.d]: Done.

Comment (nordlow) [reply druntime/src/core/exception.d]: Ahh, this opens a can of worms of error messages. I didn't think of that.

Comment (nordlow) [reply druntime/src/core/exception.d]: Green now.

**Review by RazvanN7 — APPROVED** [review]

**Pull request merged**